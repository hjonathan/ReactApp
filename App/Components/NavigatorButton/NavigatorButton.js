import React, {Component} from 'react';
import {
    Text,
    TouchableOpacity,
    View
} from 'react-native';
import Styles from './styles';
import PropTypes from 'prop-types';
import * as Res from '../../Assets/resources';
import Icon from 'react-native-vector-icons/FontAwesome';

export default class NavigatorButton extends Component {
    /**
     * Costructor
     * @param {*} props
     */
    constructor(props) {
        super(props);
    }

    /**
     * Creates the component that must be setted into the modal element.
     */
    _renderNavigatorButton = () => {
        const {
            title,
            icon,
            borderWidth
        } = this.props;
        return (
            <TouchableOpacity onPress={this.props.onPress}>
                <View style={Styles.navTitleBox}>
                    <View style={Styles.navTitle}>
                        <Icon name={icon} style={{color: Res.colors.white, fontSize: 24}}/>
                    </View>
                    <View style={Styles.navOption}>
                        <Text style={Styles.navTitleText}>{title}</Text>
                    </View>
                </View>
            </TouchableOpacity>
        );
    };

    /**
     * Render and create the modal element
     */
    render() {
        return (
            this._renderNavigatorButton()
        );
    }
}
/**
 * Initialize the prop to be mapped and validated into the component
 *
 * @ignore
 */
NavigatorButton.propTypes = {
    title: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired,
    onPress: PropTypes.func.isRequired,
    borderWidth: PropTypes.number,
}
/**
 * Sets the default values
 */
NavigatorButton.defaultProps = {
    title: '',
    icon: '',
    borderWidth: 1
}