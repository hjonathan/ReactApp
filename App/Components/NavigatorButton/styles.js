import {StyleSheet} from 'react-native';

export default StyleSheet.create({
    navTitleText: {
        color: 'white',
        margin: 5
    },
    navTitle: {
        paddingLeft: 15
    },
    navOption: {
        justifyContent: "flex-start",
        paddingLeft: 18
    },
    navCounter: {
        flex: 1,
    },
    navTitleBox: {
        flexDirection: 'row',
        flex: 1,
        paddingTop: 5,
        paddingBottom: 10
    }
});