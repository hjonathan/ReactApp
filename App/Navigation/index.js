import {connect} from "react-redux";
import Actions from "../Actions";
import AppNavigation from "./navigation";

const mapStateProps = state => ({
    navigationState: state.NavigationReducer,
    isLoggedIn: state.Oauth.isLoggedIn || false,
    isUploadCase: state.Net.isUploadCase || false,
    routeName: state.Cases.routeName
});

const mapDispatchToProps = dispatch => ({
    connectionChange: isConnected => dispatch(Actions.net.connected.update({isConnected})),
    syncUp: () => dispatch(Actions.net.syncUp.request()),
    uploadCase: status => dispatch(Actions.net.syncUp.upload(status)),
    dispatch
});

export default connect(mapStateProps, mapDispatchToProps)(AppNavigation);
