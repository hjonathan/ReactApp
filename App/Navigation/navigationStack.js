import React from "react";
import {connect} from "react-redux";
import {Image, I18nManager} from "react-native";
import {
    createStackNavigator,
    createDrawerNavigator,
    createSwitchNavigator
} from "react-navigation";
import {
    reduxifyNavigator,
    createReactNavigationReduxMiddleware
} from "react-navigation-redux-helpers";
import * as Res from "../Assets/resources";
import WalkThrough from "../Screens/WalkThrough";
import Login from "../Screens/Login";
import FormRender from "../Screens/FormRender";
import Inbox from "../Screens/Inbox";
import Draft from "../Screens/Draft";
import Unassigned from "../Screens/Unassigned";
import Participated from "../Screens/Participated";
import Switch from "../Screens/Switch";
import Settings from "../Screens/Settings";
import CaseInformation from "../Screens/CaseInformation";
import CaseNotes from "../Screens/CaseNotes";
import History from "../Screens/History";
import NewCases from "../Screens/NewCases";
import NavigatorTitle from "../Components/NavigatorTitle";
import I18n from "../I18n";
import BodyDrawer from "./bodyDrawer";
import {CASE} from "../Libs/Const";

/**
 * Define the navigator stack with all available screens into the App.
 */
const InboxStack = createStackNavigator({
        Inbox,
        History,
        CaseNotes,
        NewCases,
        FormRender,
        Settings
    }, {
        initialRouteName: CASE.INBOX
    }),
    UnassignedStack = createStackNavigator({
        Unassigned,
        NewCases,
        FormRender,
        CaseNotes,
        History,
        CaseInformation,
        Settings
    }, {
        initialRouteName: CASE.UNASSIGNED
    }),
    ParticipatedStack = createStackNavigator({
        Participated,
        NewCases,
        CaseNotes,
        History,
        CaseInformation,
        Settings
    }, {
        initialRouteName: CASE.PARTICIPATED
    }),
    DraftStack = createStackNavigator({
        Draft,
        NewCases,
        FormRender,
        CaseNotes,
        History,
        Settings
    }, {
        initialRouteName: CASE.DRAFT
    }),
    /**
     * Function for navigationOptions in drawer stacks items
     * @param menuName Name of list
     * @param icon Icon of list
     * @param number is number case in a list
     * @returns {{drawerLabel: *, drawerIcon: *}}
     */
    navOptionsListCase = (menuName, icon, number) => ({
        drawerLabel: (
            <NavigatorTitle
                title={menuName}
                number={number}
            />
        ),
        drawerIcon: (
            <Image
                source={icon}
                style={{width: 24, height: 24}}
            />
        )
    }),
    /**
     * Define the draw navigator stack with all available App.
     */
    AppDrawerStack = createDrawerNavigator({
        Inbox: {
            screen: InboxStack,
            navigationOptions: ({navigation}) => {
                const number = navigation.state.params && navigation.state.params.counters
                    ? navigation.state.params.counters.toDo
                    : 0;
                return navOptionsListCase(I18n.t("menu_drawer_main_menu_item_inbox"), Res.images.inbox, number);
            }
        },
        Unassigned: {
            screen: UnassignedStack,
            navigationOptions: ({navigation}) => {
                const number = navigation.state.params && navigation.state.params.counters
                    ? navigation.state.params.counters.unassigned
                    : 0;
                return navOptionsListCase(
                    I18n.t("menu_drawer_main_menu_item_unassigned"),
                    Res.images.unassigned,
                    number
                );
            }
        },
        Participated: {
            screen: ParticipatedStack,
            navigationOptions: ({navigation}) => {
                const number = navigation.state.params && navigation.state.params.counters
                    ? navigation.state.params.counters.participated
                    : 0;
                return navOptionsListCase(
                    I18n.t("menu_drawer_main_menu_item_participated"),
                    Res.images.participated,
                    number
                );
            }
        },
        Draft: {
            screen: DraftStack,
            navigationOptions: ({navigation}) => {
                const number = navigation.state.params && navigation.state.params.counters
                    ? navigation.state.params.counters.draft
                    : 0;
                return navOptionsListCase(I18n.t("menu_drawer_main_menu_item_draft"), Res.images.draft, number);
            }
        }
    }, {
        contentComponent: BodyDrawer,
        drawerPosition: I18nManager.isRTL ? "right" : "left"
    }),
    /**
     * Define the Login navigator stack.
     */
    LoginStack = createStackNavigator({
        Login: {
            screen: Login,
            navigationOptions: {
                header: null
            }
        },
        Settings
    }, {
        initialRouteName: "Login"
    }),
    /**
     * Define the Switch navigator stack.
     */
    NavigatorStack = createSwitchNavigator(
        {
            Auth: Switch,
            WalkThrough,
            Login: LoginStack,
            Drawer: AppDrawerStack
        },
        {
            initialRouteName: "Auth"
        }
    ),
    /**
     * Function to work navigation redux helpers
     * @returns {{navMiddleware, AppWithNavigationState}}
     */
    redux = () => {
        const navMiddleware = createReactNavigationReduxMiddleware(
                "Login",
                state => state.NavigationReducer
            ),
            AppWithNavigationState = reduxifyNavigator(NavigatorStack, "Login");
        return {
            navMiddleware,
            AppWithNavigationState
        };
    },
    {navMiddleware, AppWithNavigationState} = redux(),
    mapStateToProps = state => ({
        state: state.NavigationReducer
    }),
    AppNavigator = connect(mapStateToProps)(AppWithNavigationState);

export default NavigatorStack;
export {AppNavigator, navMiddleware};
