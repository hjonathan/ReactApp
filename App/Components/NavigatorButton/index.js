import NavigatorButton from './NavigatorButton';
export default NavigatorButton;