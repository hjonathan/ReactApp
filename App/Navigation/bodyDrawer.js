import React, {Component} from "react";
import {TouchableOpacity, View, Text, Image, ScrollView, Linking} from "react-native";
import {
    DrawerItems,
    NavigationActions
} from "react-navigation";
import Modal from "react-native-modal";
import Actions from "../Actions";
import Styles from "./styles";
import NavigatorButton from "../Components/NavigatorButton/NavigatorButton";
import * as Res from "../Assets/resources";
import I18n from "../I18n";
import config from "../conf.json";

/**
 * Body for drawer option (Inbox, Draft, Unassigned, Participated)
 */
class BodyDrawer extends Component {
    constructor (props) {
        super(props);
        this.state = {
            modalVisible: false,
            modalVisibleLogOut: false
        };
    }

    /**
     * Close modal handler
     */
    closeModal () {
        this.setState({modalVisible: false});
    }

    /**
     * Close Modal Logout handler.
     */
    closeModalLogOut () {
        this.setState({modalVisibleLogOut: false});
    }

    /**
     * Close Body Drawer.
     */
    closeBodyDrawer () {
        this.props.navigation.closeDrawer();
    }
    /**
     * Renders header
     * @returns {*}
     */
    renderHeader () {
        let userData = {};
        if (this.props.navigation.state.params && this.props.navigation.state.params.userData) {
            userData = this.props.navigation.state.params.userData;
        }
        return (
            <View style={Styles.contentHeader}>
                <Image
                    style={Styles.avatar}
                    source={{uri: `data:image/gif;base64,${userData.userPhoto ? userData.userPhoto : null}`}}
                />
                <Text style={Styles.userInfo}>
                    {userData.firstName && userData.lastName ? `${userData.firstName} ${userData.lastName}` : null}
                </Text>
            </View>
        );
    }

    /**
     * Renders NavButtons subItems
     * @returns {*}
     */
    renderSubItems () {
        return (
            <View>
                <NavigatorButton
                    title={I18n.t("menu_drawer_settings_item_settings")}
                    icon="cog"
                    onPress={() => {
                        const settings = NavigationActions.navigate({
                            routeName: "Settings",
                            params: {
                                logged: true
                            }
                        });
                        this.props.navigation.dispatch(settings);
                    }}
                />
                <NavigatorButton
                    title={I18n.t("menu_drawer_settings_item_help")}
                    icon="question-circle"
                    onPress={() => {
                        Linking.openURL(config.helpUrl);
                    }}
                />
                <NavigatorButton
                    title={I18n.t("manu_drawer_settings_item_about")}
                    icon="info-circle"
                    onPress={() => {
                        this.setState({modalVisible: true});
                    }}
                />
                <NavigatorButton
                    title={I18n.t("menu_drawer_settings_item_Logout")}
                    icon="sign-out"
                    onPress={() => {
                        this.closeBodyDrawer();
                        this.setState({modalVisibleLogOut: true});
                    }}
                />
            </View>
        );
    }

    /**
     * Renders the information modal
     * @returns {*}
     */
    renderModal () {
        return (
            <Modal
                isVisible={this.state.modalVisible}
                animationType="slide"
                backdropOpacity={0.6}
                onRequestClose={() => this.closeModal()}
            >
                <View style={Styles.modalContainer}>
                    <View style={Styles.innerContainer}>
                        <View style={Styles.modalTitle}>
                            <Text style={Styles.modalTextTitle}>{config.application.name}</Text>
                        </View>
                        <View style={{padding: 16}}>
                            <Text style={Styles.modalTextContainer}>
                                {I18n.t("dialog_about_version_label")} {config.version.number}
                            </Text>
                            <Text style={Styles.modalTextContainer}>
                                {I18n.t("dialog_about_build_name_label")} {config.version.gitHash}
                            </Text>
                        </View>
                        <View style={Styles.modalFooter}>
                            <View style={Styles.buttonContainer}>
                                <TouchableOpacity
                                    style={Styles.okButton}
                                    onPress={() => {
                                        this.closeModal();
                                    }}
                                >
                                    <Text style={{color: "white"}}>
                                        {I18n.t("OK").toUpperCase()}
                                    </Text>
                                </TouchableOpacity>
                            </View>
                        </View>
                    </View>
                </View>
            </Modal>
        );
    }

    /**
     * Render modal LogOut
     * @returns {*}
     */
    renderModalLogOut () {
        return (
            <Modal
                isVisible={this.state.modalVisibleLogOut}
                backdropOpacity={0.6}
                animationType="none"
            >
                <View style={Styles.logOutContainer}>
                    <View style={Styles.logOutSubContainer}>
                        <View style={Styles.logOutHeader}>
                            <Text style={{color: Res.colors.white, fontSize: 18}}>
                                {I18n.t("menu_drawer_settings_item_Logout")}
                            </Text>
                        </View>
                        <View style={Styles.logOutBody}>
                            <Text>
                                {I18n.t("AreLogout")}
                            </Text>
                        </View>
                        <View style={Styles.logOutFooter}>
                            <View style={{width: "50%"}}>
                                <TouchableOpacity
                                    style={Styles.cancelButton}
                                    onPress={() => {
                                        this.closeModalLogOut();
                                    }}
                                >
                                    <Text style={{color: "white"}}>
                                        {I18n.t("Cancel").toUpperCase()}
                                    </Text>
                                </TouchableOpacity>
                            </View>
                            <View style={{width: "50%"}}>
                                <TouchableOpacity
                                    style={Styles.okButton}
                                    onPress={() =>
                                        this.props.navigation.dispatch(Actions.logout.request())
                                    }
                                >
                                    <Text style={{color: "white"}}>
                                        {I18n.t("OK").toUpperCase()}
                                    </Text>
                                </TouchableOpacity>
                            </View>
                        </View>
                    </View>
                </View>
            </Modal>
        );
    }

    /**
     * Renders sidebar
     * @returns {*}
     */
    render () {
        return (
            <View style={{flex: 1, backgroundColor: Res.colors.bgColor}}>
                <ScrollView
                    contentContainerStyle={{flexGrow: 1}}
                    keyboardShouldPersistTaps="always"
                    keyboardDismissMode="on-drag"
                >
                    {this.renderHeader()}
                    <View style={Styles.divider} />
                    <DrawerItems
                        {...this.props}
                        labelStyle={{color: "#ffffff"}}
                        onItemPress={({route, focused}) => {
                            // here we overwrite the function with a close body drawer
                            // + the original function
                            this.closeBodyDrawer();
                            this.props.onItemPress({route, focused});
                        }}
                    />
                    <View style={Styles.divider} />
                    {this.renderSubItems()}
                </ScrollView>
                {this.renderModal()}
                {this.renderModalLogOut()}
            </View>
        );
    }
}

export default BodyDrawer;
