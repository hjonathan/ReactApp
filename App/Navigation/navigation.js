import React, {Component} from "react";
import {Platform, StatusBar, I18nManager, NetInfo, BackHandler} from "react-native";
import {NavigationActions} from "react-navigation";
import queueFactory from "react-native-queue";
import _ from "lodash";
import * as Res from "../Assets/resources";
import {AppNavigator} from "./navigationStack";
import Utils from "../Utils";
import Actions from "../Actions";
import I18n from "../I18n";
import defaultPreferences from "../conf.json";

/**
 * Class App Navigation
 */
class AppNavigation extends Component {
    constructor (props) {
        super(props);
        this.state = {
            queue: null
        };
        this.init();
    }

    /**
     * Gets current language and set default language
     */
    componentWillMount () {
        this.initPreferences();
    }

    /**
     * Gets current platform OS and version
     * then sets StatusBar Color only on Android
     */
    componentDidMount () {
        if (Platform.OS === "android" && Platform.Version >= 21) {
            StatusBar.setBackgroundColor(Res.colors.darkColor);
        }
        NetInfo.isConnected.addEventListener("connectionChange", this.handleConnectionChange);
        BackHandler.addEventListener("hardwareBackPress", this.onBackPress);
    }

    /**
     * Component is being removed
     */
    componentWillUnmount () {
        NetInfo.isConnected.removeEventListener("connectionChange", this.handleConnectionChange);
        BackHandler.removeEventListener("hardwareBackPress", this.onBackPress);
    }

    /**
     * Is press Back button
     * @returns {boolean}
     */
    onBackPress = () => {
        const {dispatch, navigationState, routeName} = this.props;
        let objectRoute,
            routes;
        if (navigationState.index === 0) {
            return false;
        }
        if (routeName) {
            objectRoute = Utils.findRouteByRouteName(navigationState, routeName);
            routes = objectRoute ? _.last(objectRoute.routes) : {};
            if (Object.prototype.hasOwnProperty.call(routes, "routeName") &&
                routes.routeName === "FormRender") {
                dispatch(Actions.formRender.backButtonPressed());
                return true;
            }
        }
        dispatch(NavigationActions.back());
        return true;
    };

    /**
     * Initializes user settings
     */
    async initPreferences () {
        const serverUrl = await Utils.getItem("@server"),
            workspace = await Utils.getItem("@workspace"),
            language = await Utils.getItem("@lang");

        if (!serverUrl) {
            Utils.setItem("@server", defaultPreferences.server);
        }
        if (!workspace) {
            Utils.setItem("@workspace", defaultPreferences.workspace);
        }
        if (!language) {
            Utils.setItem("@lang", defaultPreferences.lang);
        }
        I18n.defaultLocale = language || defaultPreferences.lang;
        I18n.locale = language || defaultPreferences.lang;
        I18nManager.forceRTL(I18n.locale === "ar");
    }

    /**
     * Initialized queue parameters
     * @returns {Promise<void>}
     */
    async init () {
        const queue = await queueFactory();
        // Job for sync with server
        queue.addWorker("sync-up", async (id, payload) => {
            await this.props.syncUp();
        });
        // Start queue to process any jobs that hadn't finished when app was last closed.
        queue.start();
        // Attach initialized queue to state.
        this.setState({
            queue
        });
    }

    /**
     * Change the status connection
     * @param isConnected
     */
    handleConnectionChange = async (isConnected) => {
        let existMainFile = await Utils.wasCleaned();
        this.props.connectionChange(isConnected);
        if (isConnected && this.props.isLoggedIn && !this.props.isUploadCase) {
            this.props.uploadCase(true);
            this.state.queue.createJob("sync-up");
            if (!existMainFile) {
                this.props.dispatch(Actions.fs.app.buildProd.download());
            }
        }
    };

    /**
     * Renders
     * @returns {*}
     */
    render () {
        return (
            <AppNavigator />
        );
    }
}

export default AppNavigation;
