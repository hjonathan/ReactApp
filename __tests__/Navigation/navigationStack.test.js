import React from "react";
import {TouchableOpacity} from "react-native";
import Modal from "react-native-modal";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import configureStore from "redux-mock-store";
import Navigation from "../../App/Navigation";
import AppNavigation from "../../App/Navigation/navigation";
import NavigatorStack from "../../App/Navigation/navigationStack";
import BodyDrawer from "../../App/Navigation/bodyDrawer";
import NavigatorButton from "../../App/Components/NavigatorButton/NavigatorButton";

jest.mock('react-native-i18n', () => require.requireActual('../../__mocks__/react-native-i18n').default);

describe("Navigation", () => {
    const initialState = {
        Oauth: {
            isLoggedIn: true
        },
        Net: {
            isUploadCase: false
        },
        Cases: {
            routeName: "Test"
        }
    };
    const mockStore = configureStore(),
        dispatch = () => {
        },
        navigationState = {
            index: 0,
            routes: [
                {key: "WalkThrough", routeName: "WalkThrough"}
            ]
        },
        connectionChange = jest.fn(),
        uploadCase = jest.fn(),
        navigation = {
            dispatch: jest.fn(),
            state: {
                params: {
                    userData: null
                }
            },
            closeDrawer: jest.fn()
        };
    let store,
        connect,
        viewAppNavigation,
        navigationStack,
        viewBodyDrawer;

    beforeEach(() => {
        store = mockStore(initialState);
        // navigationReducer.reduxifyNavigator = jest.fn();
        connect = shallow(
            <Navigation
                store={store}
                dispatch={jest.fn()}
                navigationState={navigationState}
            />
        );
        viewAppNavigation = shallow(
            <AppNavigation
                store={store}
                dispatch={jest.fn()}
                navigationState={navigationState}
                connectionChange={connectionChange}
                uploadCase={uploadCase}
            />
        );
        viewBodyDrawer = shallow(<BodyDrawer navigation={navigation} />);
        navigationStack = shallow(
            <NavigatorStack
                navigation={{dispatch, state: navigationState}}
            />
        );
        viewBodyDrawer.setState({modalVisible: false});
        viewBodyDrawer.setState({modalVisibleLogOut: false});
    });

    /**
     * Render test
     */
    test('Render connect navigation', () => {
        expect(toJson(connect)).toMatchSnapshot();
    });

    test('mapStateToProps', () => {
        expect(connect.props('children').connectionChange()).toMatchSnapshot();
        expect(connect.props('children').syncUp()).toMatchSnapshot();
        expect(connect.props('children').uploadCase()).toMatchSnapshot();
    });

    test("Props coverage componentWillMount", () => {
        expect(viewAppNavigation.instance().componentWillMount()).toBeUndefined();
    });

    test("Props coverage componentDidMount", () => {
        expect(viewAppNavigation.instance().componentDidMount()).toBeUndefined();
    });

    test("Props coverage componentWillUnmount", () => {
        expect(viewAppNavigation.instance().componentWillUnmount()).toBeUndefined();
    });

    test("Props coverage initPreferences", async () => {
        expect(viewAppNavigation.instance().onBackPress()).toEqual(false);
        viewAppNavigation.setProps({navigationState: {index: 2}});
        expect(viewAppNavigation.instance().onBackPress()).toEqual(true);
        viewAppNavigation.setProps({
            navigationState: {
                index: 2,
                routes: []
            },
            routeName: "Test"
        });
        expect(viewAppNavigation.instance().onBackPress()).toEqual(true);
    });

    test("Props coverage initPreferences", async () => {
        expect(await viewAppNavigation.instance().initPreferences()).toBeUndefined();
    });

    test("Props coverage init", async () => {
        expect(await viewAppNavigation.instance().init()).toBeUndefined();
    });

    test("Props coverage handleConnectionChange", async () => {
        viewAppNavigation.setProps({isLoggedIn: true, isUploadCase: false});
        viewAppNavigation.setState({queue: {createJob: jest.fn()}});
        expect(await viewAppNavigation.instance().handleConnectionChange(true)).toBeUndefined();
    });

    test("Props coverage BodyDrawer >> closeModal", () => {
        expect(viewBodyDrawer.instance().closeModal()).toBeUndefined();
    });

    test("Props coverage BodyDrawer >> closeModalLogOut", () => {
        expect(viewBodyDrawer.instance().closeModalLogOut()).toBeUndefined();
    });

    test("Props coverage BodyDrawer >> closeBodyDrawer", () => {
        expect(viewBodyDrawer.instance().closeBodyDrawer()).toBeUndefined();
    });

    test("Props coverage BodyDrawer >> renderHeader", () => {
        viewBodyDrawer.setProps({
            navigation: {
                state: {
                    params: {
                        userData: {
                            userPhoto: "picture",
                            lastName: "Test",
                            firstName: "Test"
                        }
                    }
                }
            }
        });
        expect(toJson(viewBodyDrawer.instance().renderHeader())).toMatchSnapshot();
    });

    test("Props coverage BodyDrawer >> renderSubItems", () => {
        const viewJson = toJson(viewBodyDrawer.instance().renderSubItems());
        expect(viewJson).toMatchSnapshot();
        expect(viewBodyDrawer.find(NavigatorButton).length).toEqual(4);
        viewBodyDrawer.find(NavigatorButton).at(0).props().onPress();
        viewBodyDrawer.find(NavigatorButton).at(1).props().onPress();
        viewBodyDrawer.find(NavigatorButton).at(2).props().onPress();
        viewBodyDrawer.find(NavigatorButton).at(3).props().onPress();
    });

    test("Props coverage BodyDrawer >> renderModal", () => {
        viewBodyDrawer.setState({modalVisible: true});
        const viewJson = toJson(viewBodyDrawer.instance().renderModal());
        expect(viewJson).toMatchSnapshot();
        expect(viewBodyDrawer.find(TouchableOpacity).length).toEqual(3);
        viewBodyDrawer.find(TouchableOpacity).at(0).props().onPress();
        viewBodyDrawer.find(Modal).first().simulate('RequestClose')
    });

    test("Props coverage BodyDrawer >> renderModalLogOut", () => {
        viewBodyDrawer.setState({modalVisibleLogOut: true});
        let logOutModal = viewBodyDrawer.instance().renderModalLogOut(),
            footer = logOutModal.props.children.props.children.props.children[2],
            buttonCancel = footer.props.children[0].props.children,
            buttonOk = footer.props.children[1].props.children;
        expect(buttonOk.props.onPress()).toMatchSnapshot();
        expect(buttonCancel.props.onPress()).toMatchSnapshot();
        expect(logOutModal).toMatchSnapshot();
        viewBodyDrawer.instance().setState({modalVisibleLogOut: false});
        expect(logOutModal).toMatchSnapshot();
    });

    test('Render stack navigation', () => {
        expect(toJson(NavigatorStack)).toMatchSnapshot();
    });
});
