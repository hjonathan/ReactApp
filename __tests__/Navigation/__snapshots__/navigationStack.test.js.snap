// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Props coverage BodyDrawer >> renderHeader 1`] = `undefined`;

exports[`Navigation Props coverage BodyDrawer >> renderModal 1`] = `undefined`;

exports[`Navigation Props coverage BodyDrawer >> renderModalLogOut 1`] = `undefined`;

exports[`Navigation Props coverage BodyDrawer >> renderModalLogOut 2`] = `undefined`;

exports[`Navigation Props coverage BodyDrawer >> renderModalLogOut 3`] = `
<ReactNativeModal
  animationIn="slideInUp"
  animationInTiming={300}
  animationOut="slideOutDown"
  animationOutTiming={300}
  animationType="none"
  avoidKeyboard={false}
  backdropColor="black"
  backdropOpacity={0.6}
  backdropTransitionInTiming={300}
  backdropTransitionOutTiming={300}
  hideModalContentWhileAnimating={false}
  isVisible={true}
  onBackButtonPress={[Function]}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalShow={[Function]}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  useNativeDriver={false}
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Component
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "center",
          "width": "75%",
        }
      }
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#3397E1",
            "height": 48,
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#FFF",
              "fontSize": 18,
            }
          }
        >
          menu_drawer_settings_item_Logout
        </Text>
      </Component>
      <Component
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFF",
            "height": 48,
            "justifyContent": "center",
            "padding": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          AreLogout
        </Text>
      </Component>
      <Component
        style={
          Object {
            "flexDirection": "row",
            "height": 48,
            "justifyContent": "flex-end",
          }
        }
      >
        <Component
          style={
            Object {
              "width": "50%",
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#E14333",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                }
              }
            >
              CANCEL
            </Text>
          </TouchableOpacity>
        </Component>
        <Component
          style={
            Object {
              "width": "50%",
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#1FBC99",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                }
              }
            >
              OK
            </Text>
          </TouchableOpacity>
        </Component>
      </Component>
    </Component>
  </Component>
</ReactNativeModal>
`;

exports[`Navigation Props coverage BodyDrawer >> renderModalLogOut 4`] = `
<ReactNativeModal
  animationIn="slideInUp"
  animationInTiming={300}
  animationOut="slideOutDown"
  animationOutTiming={300}
  animationType="none"
  avoidKeyboard={false}
  backdropColor="black"
  backdropOpacity={0.6}
  backdropTransitionInTiming={300}
  backdropTransitionOutTiming={300}
  hideModalContentWhileAnimating={false}
  isVisible={true}
  onBackButtonPress={[Function]}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalShow={[Function]}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  useNativeDriver={false}
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Component
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "center",
          "width": "75%",
        }
      }
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#3397E1",
            "height": 48,
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#FFF",
              "fontSize": 18,
            }
          }
        >
          menu_drawer_settings_item_Logout
        </Text>
      </Component>
      <Component
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFF",
            "height": 48,
            "justifyContent": "center",
            "padding": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          AreLogout
        </Text>
      </Component>
      <Component
        style={
          Object {
            "flexDirection": "row",
            "height": 48,
            "justifyContent": "flex-end",
          }
        }
      >
        <Component
          style={
            Object {
              "width": "50%",
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#E14333",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                }
              }
            >
              CANCEL
            </Text>
          </TouchableOpacity>
        </Component>
        <Component
          style={
            Object {
              "width": "50%",
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#1FBC99",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                }
              }
            >
              OK
            </Text>
          </TouchableOpacity>
        </Component>
      </Component>
    </Component>
  </Component>
</ReactNativeModal>
`;

exports[`Navigation Props coverage BodyDrawer >> renderSubItems 1`] = `undefined`;

exports[`Navigation Render connect navigation 1`] = `
<AppNavigation
  connectionChange={[Function]}
  dispatch={[Function]}
  isLoggedIn={true}
  isUploadCase={false}
  routeName="Test"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  syncUp={[Function]}
  uploadCase={[Function]}
/>
`;

exports[`Navigation Render stack navigation 1`] = `undefined`;

exports[`Navigation mapStateToProps 1`] = `
Object {
  "payload": Object {
    "isConnected": undefined,
  },
  "type": "net/connected/update",
}
`;

exports[`Navigation mapStateToProps 2`] = `
Object {
  "payload": Object {
    "status": undefined,
  },
  "type": "net/syncUp/request",
}
`;

exports[`Navigation mapStateToProps 3`] = `
Object {
  "type": "net/syncUp/upload",
}
`;
